### ISSUE TITLE
Custom Snippets for Twig not loaded

### ISSUE THREAD

**Issue by fvwanja** [microsoft/vscode#226053]
Type: <b>Bug</b>

JSON files containing snippets for various languages such as JavaScript, PHP and Twig are being loaded. This used to work until approx. 2 months ago. Now the twig.json file is not being loaded with .twig files. I can create a global twig.code-snippets file, which works – so the JSON must be valid (also it did work before). It seems that the filename – which represents the file extension „twig” – is no longer associated with the file format.

When I run Shift + CMD + P and select „Snippets: Insert Snippet”, the list does not contain my custom snippets from twig.json.

When I run Shift + CMD + P and select „Snippets: Configure Snippets” and type „twig”, I can select the twig.json file and edit it, but upon selecting the file, it sais „twig.json (null)” (see screenshot). Clearly .twig files are not associated with the Twig language.

![Screenshot 2024-08-20 at 14 53 45](https://github.com/user-attachments/assets/d2736307-6ca8-4661-8d12-99a6bc349640)

VS Code version: Code 1.92.2 (Universal) (fee1edb8d6d72a0ddff41e5f71a671c23ed924b9, 2024-08-14T17:29:30.058Z)
OS version: Darwin arm64 23.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 3, 3|
|Memory (System)|16.00GB (0.10GB free)|
|Process Argv|. --crash-reporter-id 0550cbce-34d0-4ec2-8be9-8f3905f03f5b|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
npm-intellisense|chr|1.4.5
vscode-eslint|dba|3.0.10
vscode-quick-select|dba|0.2.9
composer-php-vscode|DEV|1.50.15872
intelli-php-vscode|DEV|0.12.15062
phptools-vscode|DEV|1.50.15872
profiler-php-vscode|DEV|1.50.15872
gitlens|eam|15.3.0
EditorConfig|Edi|0.16.4
auto-rename-tag|for|0.1.10
shell-format|fox|7.2.5
vscode-env|Iro|0.1.0
twig-language|mbl|0.9.4
vscode-typescript-next|ms-|5.6.20240819
vs-sequential-number|nep|1.1.0
material-icon-theme|PKi|5.9.0
nbsp-vscode|pos|0.0.1
vscode-sassdoc|raf|1.0.3
shellman|Rem|5.7.0
some-sass|Som|3.5.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
724cj586:31013169
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
newcmakeconfigv2:31071590
impr_priority:31102340
refactort:31108082
pythonrstrctxt:31112756
wkspc-onlycs-t:31111718
wkspc-ranged-c:31118571

```

</details>

**Comment by jrieken** [comment]
> Clearly .twig files are not associated with the Twig language.

I have pushed a change to fix the `null` but it isn't clear to me if there is anything else here. @fvwanja Can you clarify? Does the rest work and is this just about the label?

**Comment by fvwanja** [comment]
Hi @jrieken ,

thank you for your reply. I have the latest version of VS Code installed (when I can do anything else to test your submitted Code, please let me know how I can do that):

Version: 1.96.0 (Universal)
Commit: 138f619c86f1199955d53b4166bef66ef252935c
Date: 2024-12-11T02:29:09.626Z
Electron: 32.2.6
ElectronBuildId: 10629634
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Darwin arm64 23.5.0

![Image](https://github.com/user-attachments/assets/1b44fb04-caa4-4405-9977-f0336c70c72d)

When I hit Shift + CMD + P and enter „Sippets: Configure Snippets” I have a list of suggestions like `azcli`, `Batch`, `bats`, etc. `Twig` is not included in this list (see screenshot above).

I therefore create a file `~/Library/Application\ Support/Code/User/snippets/twig.json` manually and fill it with some contents like:

```json
{
  "ifblock": {
    "prefix": "ifblock",
    "body": [
      "{% if ${1:fieldName} %}",
      "\t<div class=\"${1:fieldName}\">",
      "\t\t{{ ${1:fieldName} }}",
      "\t</div>",
      "{% endif %}"
    ],
    "description": "Generate unordered list with for"
  }
}
```

I can again confirm that the configuration of the `twig.json` file is correct, because it works, when I put the same contents into a global file `twig.code-snippets` (as I have mentioned [earlier](https://github.com/microsoft/vscode/issues/226053#issue-2475579331)).

I would expect to now be able to type `ifblock` in any .twig file and use this custom snippet. But it does not show up and the result for the action described in my previous comment https://github.com/microsoft/vscode/issues/226053#issue-2475579331 remains the same. I still see Twig listed, but assigned to the language `null`.

Summarized:

I cannot use custom snippets for .twig files, unless I put them into a global file – with the downturn of having Twig snippets in any other language.

My guess is, that VS Code is trying to assign a language to .twig files – but fails to do so – when it should only just load the custom snippets for `*.twig` files, when I work on any `.twig` file.

So I hope this helps. If I can be of further help, let me know.

**Comment by fvwanja** [comment]
Alright I installed `Version: 1.97.0-insider` and I am now seeing `twig (twig)` and not `twig (null)` anymore (🥳 yay).

![Image](https://github.com/user-attachments/assets/6cf3b47e-e131-493a-ba5f-104d50de31fb)

However I am still unable to use Snippets from `~/Library/Application\ Support/Code\ -\ Insiders/User/snippets/twig.json` in .twig files.

With no extensions installed, the language indicator on the lower right reads `Plain Text`. I guess Twig is just not natively supported as a language in VS Code.

![Image](https://github.com/user-attachments/assets/6ebcfd36-22af-490e-89dc-5a98072d441a)

I tried 2 extensions:
- `mblode.twig-language` changes the language indicator to `HTML`
- `mblode.twig-language-2` changes the language indicator to `HTML (Twig)`

The custom snippets from the `twig.json` file work, when the language indicator shows `HTML (Twig)` (🥳 yay).

So to summarize this:

VS Code cannot load custom snippets for a file extension, when it cannot associate it with a language. It requires a third party extension like [mblode.twig-language-2](https://marketplace.visualstudio.com/items?itemName=mblode.twig-language-2).

In my personal opinion, custom snippets should also be associated with the file extension – but there might just be a reason, why they are not.

So this does at least settle my issue. Thank you for your time and effort! 🙏
